src/beast/beast/container/vectum.h

```diff
@@ -0,0 +1,642 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of Beast: https://github.com/vinniefalco/Beast
+    Copyright 2014, Vinnie Falco <vinnie.falco@gmail.com>
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_CONTAINER_VECTUM_H_INCLUDED
+#define BEAST_CONTAINER_VECTUM_H_INCLUDED
+
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+#include <iterator>
+#include <memory>
+#include <new>
+#include <beast/cxx14/type_traits.h> // <type_traits>
+#include <utility>
+
+namespace beast {
+namespace nudb {
+namespace detail {
+
+template <class>
+class vectum;
+
+template <class T>
+void
+swap (vectum<T>&, vectum<T>&);
+
+// Custom vector where the element size is controlled
+// at run-time during construction and can be larger than
```

Comment:
So this isn't quite clear from the description.

`value_type`s are stored in each entry, and there's extra memory left over - so far so good.  Is this per entry?  Can you resize individual entries or must you resize the whole thing?

What about performance properties?  How fast does it insert, how much memory does it use, how fast does it do the "element size" resizing?

Once I have answers to these questions, I'll be able to suggest a better name than `vectum` which will make us the laughing stock of the industry ("Vinnie's vectum").